extensions/azurecore/src/account-provider/auths/azureAuthCodeGrant.ts

```diff
@@ -0,0 +1,224 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as crypto from 'crypto';
+import * as nls from 'vscode-nls';
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import * as http from 'http';
+
+import {
+	AzureAuth,
+	AccessToken,
+	RefreshToken,
+	TokenClaims,
+	TokenRefreshResponse,
+} from './azureAuth';
+
+import {
+	AzureAccountProviderMetadata,
+	Tenant,
+	Resource,
+	AzureAuthType
+} from '../interfaces';
+
+import { SimpleWebServer } from './simpleWebServer';
+import { SimpleTokenCache } from '../simpleTokenCache';
+const localize = nls.loadMessageBundle();
+
+interface Deferred<T> {
```

Comment:
looks like this is declared twice in this PR.  There isn't a good place to consolidate?

Reply:
Resolved